metabox/metabox/core/lxd_provider.py

```diff
@@ -61,7 +62,8 @@ def __init__(self, session_config, effective_machine_config,
             # TODO: Find a suitable timeout value here
             self.client = pylxd.Client(timeout=None)
         except ClientConnectionFailed as exc:
-            raise SystemExit from exc
+            logger.exception("Cannot connect to LXD. Is it installed?")
+            sys.exit()
```

Comment:
This shouldn't exit with a 0 retcode.

What I recommend instead is:
```
msg = f"Cannot connect to LXD. {exc}. Is it installed?"
logger.exception(msg)
raise SystemExit(msg) from exc
```

Reply:
The original code did not bubble up any error message. Metabox would just stop without any info. But as you said, `sys.exit()` is not any better!

The change I made should bubble up the info and stop the program. [`ClientConnectionFailed`](https://pylxd.readthedocs.io/en/latest/api.html?highlight=ClientConnectionFailed#pylxd.exceptions.ClientConnectionFailed) does not contain any response, so I just `raise SystemExit(msg)`.